src/Rector/v12/v0/typo3/HintNecessaryUploadedFileChangesRector.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Ssch\TYPO3Rector\Rector\v12\v0\typo3;
+
+use PhpParser\Comment;
+use PhpParser\Node;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\ClassMethod;
+use PhpParser\Node\UnionType;
+use Rector\Core\Console\Output\RectorOutputStyle;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @changelog https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Breaking-97214-UseUploadedFileObjectsInsteadOf_FILES.html
+ * @see \Ssch\TYPO3Rector\Tests\Rector\v12\v0\typo3\HintNecessaryUploadedFileChangesRector\HintNecessaryUploadedFileChangesRectorTest
+ */
+final class HintNecessaryUploadedFileChangesRector extends AbstractRector
+{
+    /**
+     * @var string
+     */
+    private const MESSAGE = 'When extending the Core ResourceStorage, the addUploadedFile() method needs some adaptions. See Breaking-97214-UseUploadedFileObjectsInsteadOf_FILES.html for full migration advise';
+
+    /**
+     * @var string
+     */
+    private const COMMENT = '// FIXME: Rector: https://docs.typo3.org/c/typo3/cms-core/main/en-us/Changelog/12.0/Breaking-97214-UseUploadedFileObjectsInsteadOf_FILES.html';
+
+    public function __construct(
+        private readonly RectorOutputStyle $rectorOutputStyle
+    ) {
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($this->shouldSkip($node)) {
+            return null;
+        }
+
+        $affectedMethod = $node->getMethod('addUploadedFile');
+
+        if (! $affectedMethod instanceof ClassMethod) {
+            return null;
+        }
+
+        $params = $affectedMethod->getParams();
+        if ($params[0]->type instanceof UnionType) {
+            return null;
+        }
+
+        if ($params[0]->type instanceof Identifier && 'array' === $params[0]->type->name) {
+            $comments = $affectedMethod->getComments();
+            $comments = array_filter(
+                $comments,
+                static fn (Comment $comment) => ! str_starts_with($comment->getText(), '// FIXME: Rector:')
```

Comment:
with this check i would expect a test fixture which actively checks this.
What do you think?

Reply:
done